core/src/main/java/ch/cyberduck/core/DefaultHostPasswordStore.java

```diff
@@ -214,25 +202,27 @@ public void save(final Host bookmark) throws LocalAccessDeniedException {
                     credentials.getToken());
         }
         if(credentials.isOAuthAuthentication()) {
-            final String prefix = this.getOAuthPrefix(bookmark);
+            final PasswordStorePrefixService service = bookmark.getProtocol().getFeature(PasswordStorePrefixService.class);
+
+            final String prefix = service.getPrefix(bookmark);
+            final String hostname = service.getHostname(bookmark);
+            final Integer port = service.getPort(bookmark);
+            final Scheme scheme = service.getScheme(bookmark);
             if(StringUtils.isNotBlank(credentials.getOauth().getAccessToken())) {
-                this.addPassword(bookmark.getProtocol().getScheme(),
-                        bookmark.getPort(), this.getOAuthHostname(bookmark),
+                this.addPassword(scheme, port, hostname,
                         String.format("%s OAuth2 Access Token", prefix), credentials.getOauth().getAccessToken());
             }
             if(StringUtils.isNotBlank(credentials.getOauth().getRefreshToken())) {
-                this.addPassword(bookmark.getProtocol().getScheme(),
-                        bookmark.getPort(), this.getOAuthHostname(bookmark),
+                this.addPassword(scheme, port, hostname,
                         String.format("%s OAuth2 Refresh Token", prefix), credentials.getOauth().getRefreshToken());
             }
             // Save expiry
             if(credentials.getOauth().getExpiryInMilliseconds() != null) {
-                this.addPassword(this.getOAuthHostname(bookmark), String.format("%s OAuth2 Token Expiry", prefix),
+                this.addPassword(hostname, String.format("%s OAuth2 Token Expiry", prefix),
```

Comment:
Why is expiry stored under `hostname` only, whereas refresh, access and ID tokens are stored with scheme and port? Comment on this in the source code.